src/shared/conf-parser.c

```diff
@@ -563,8 +565,10 @@ int config_parse_si_size(
         assert(data);
 
         r = parse_size(rvalue, 1000, &v);
-        if (r < 0 || (uint64_t) (size_t) v != v) {
-                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse size value, ignoring: %s", rvalue);
+        if (r >= 0 && (uint64_t) (size_t) v != v)
+                r = -E2BIG;
```

Comment:
hmm, two times you used E2BIG and once ERANGE. I think ERANGE is actually the better error, as E2BIG's message talks about "argument lists" which doesn't really apply here. Any reason you used E2BIG for these two rather than ERANGE?